### PR TITLE
Added Test Project

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- Updated formula string in PeakAnnotation
- Added check in RawDataFileReader to handle blank lines at the EOF
- Made getCuratedPeaklist function public for test project
- Updated GetFiles sfd to read file by name instead of whole folder in PeakAnnotationResultExportAsMsp
- Updated delete structurefile with filename in workspace preparation
- Added testdata
- Added test project with four test

### CODE REVIEW

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                                rawData, curatedPeaklist, refinedPeaklist, adductIon, formulaResult,
+                                _param, null, _fragmentOntologyDB);
+
+                foreach (var result in results)
+                {
+                    result.TotalScore += formulaResult.TotalScore;
+                    result.Ontology = rawData.Ontology;
+                }
+
+                FragmenterResultParcer.FragmenterResultWriter($"{_projectDir}/testdata/input/test/test.sfd", results, true);
```
Comment (hechth) [inline MsFinderTest/MsfinderConsoleApp.cs:133]: If these functions are exactly the same as the ones called in the functions from the package, can we either extract a function in the package or call the package function directly?

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            }
+
+            var output = $"{_projectDir}/testdata/input/test/test.sfd";
+            var expected = $"{_projectDir}/testdata/expected/test_expected.sfd";
```
Comment (hechth) [inline MsFinderTest/MsfinderConsoleApp.cs:137]: again - paths should be platform independent

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            var expected = $"{_projectDir}/testdata/expected/out_expected.msp";
+
+            var out_hash = GetHashCode(output);
+            var expected_hash = GetHashCode(expected);
+
+            Assert.Equal(out_hash, expected_hash);
+        }
+
+        [Fact]
+        public void formualResult()
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: resolved in dec9dec

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        private readonly List<ExistFormulaQuery> _existFormulaDB;
+        private readonly List<FragmentOntology> _fragmentOntologyDB;
+        private readonly List<ChemicalOntology> _chemicalOntologies;
+
+        public MsfinderConsoleApp(ITestOutputHelper output)
+        {
+            _output = output;
+            _projectDir = Path.GetFullPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\.."));
+            _param = MsFinderIniParcer.Read($"{_projectDir}/MSFINDER.INI");
+            _rawDataList = RawDataParcer.RawDataFileReader($"{_projectDir}/testdata/input/test.msp", _param);
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: I believe this should work in mono as well.

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        }
+
+        [Fact]
+        public void PeakAnnotation()
+        {
+            new AnnotateProcess().Run($"{_projectDir}/testdata/input/test.msp", $"{_projectDir}/MSFINDER.INI",
+            $"{_projectDir}/testdata/input/out.msp");
+
+            var output = $"{_projectDir}/testdata/input/out.msp";
+            var expected = $"{_projectDir}/testdata/expected/out_expected.msp";
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: I believe this should work in mono as well.

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            var structureFile = System.IO.Directory.GetFiles($"{_projectDir}/testdata/input/test", "test.sfd");
+            if (structureFile.Length > 0)
+            {
+                FileStorageUtility.DeleteSfdFiles(structureFile);
+            }
+
+            foreach (var rawData in _rawDataList)
+            {
+                var formulaResult = MolecularFormulaFinder.GetMolecularFormulaScore(_productIonDB, _neutralLossDB, _existFormulaDB, rawData, _param);
+                if (rawData.Ms2PeakNumber <= 0 || rawData.Smiles == null || rawData.Smiles == string.Empty) return;
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: updated with the function from the package in [dec9dec](https://github.com/RECETOX/recetox-msfinder/pull/22/commits/dec9decb3b591ff6904b28fe21abb6d3c956b48f)

File: MsFinderTest/MsfinderConsoleApp.cs [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+                var structureQuery = new ExistStructureQuery(rawData.Name, rawData.InchiKey, rawData.InchiKey, new List<int>(), formulaResult.Formula, rawData.Smiles, string.Empty, 0, new DatabaseQuery());
+                if (structureQuery == null) return;
```
[the remark this answers is not in the document]
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: updated with the function from the package in [dec9dec](https://github.com/RECETOX/recetox-msfinder/pull/22/commits/dec9decb3b591ff6904b28fe21abb6d3c956b48f)

File: MsFinderTest/MsfinderConsoleApp.cs (hunk 1 above)
Comment (zargham-ahmad) [reply MsFinderTest/MsfinderConsoleApp.cs]: updated with the function from the package in [dec9dec](https://github.com/RECETOX/recetox-msfinder/pull/22/commits/dec9decb3b591ff6904b28fe21abb6d3c956b48f)

**Review by hechth — APPROVED** [review]

**Pull request merged**